package/kedro_viz/models/flowchart.py

```diff
@@ -486,14 +486,28 @@ def is_tracking_node(self):
         """Checks if the current node is a tracking data node"""
         return self.is_json_node() or self.is_metric_node()
 
+    def get_viz_metadata(self):
+        """Gets the kedro-viz metadata config specified in the catalog"""
+        metadata = getattr(self.kedro_obj, "metadata", None)
+        if not metadata:
+            return None
+        try:
+            viz_metadata = metadata["kedro-viz"]
+        except (AttributeError, KeyError):
+            return None
+        return viz_metadata
```

Comment:
Can this be replaced with the `getattr` method since it's not a nested key?

Reply:
additional comment: Would it be better to just set this as a property `viz_metadata` instead of a get method?

Reply:
i did try this but I got some weird errors, so changed it to get_ . but i agree, it's better as a property. I will ping u again, if u have a couple of min to help me with the errors that would be appreciated. let me try doing it again.